### DISCUSSION TITLE
Support general FHIR operations in the provider?

### DISCUSSION THREAD

**Discussion by loafoe** [philips-software/terraform-provider-hsdp#27] [Ideas]
Adding support for CDR onboarding I was briefly tempted to add something like `hsdp_cdr_practioner` or `hsdp_cdr_careplan` however I doubt they would see any use, unless someone can come up with a good use case of maintaining FHIR resources with Terraform?

**Comment by loafoe** [comment]
The one exception might be `hsdp_cdr_subscription` (see [STU3:Subscription](https://www.hl7.org/fhir/stu3/subscription.html)) which takes `rest-hooks` as input. Terraform might be used to manage the REST hook endpoint application, which would make it trivial to manage the configuration in CDR. What do you think?

**Comment by RaymondPhilips** [comment]
At least from our perspective i don't see a use of provisioning any other resources than Organization by terraform. Practitioner should be strongly linked to an IAM user, which we don't provision but are created on the fly through federation to the hospital's IDP.
But different projects may have different perspectives.
For our use case the only configuration done through terraform will be the creation of a root Organization resource per tenant.

**Reply by loafoe** [reply]
Thanks @RaymondPhilips for your insights. I agree, I can't see any other use-case at this time except Org onboarding and [hsdp_cdr_subscription](https://registry.terraform.io/providers/philips-software/hsdp/latest/docs/resources/cdr_subscription) callback URL management. Looking forward to your feedback on use of the resources 👍🏻

**Reply by loafoe** [reply]
@RaymondPhilips interestingly, due to the strong link to IAM identity of practitioners there is now a request to add it. There are use cases where IAM users are (fully) managed by Terraform, making it very attractive to do the same for the linked Practioners.

**Reply by RaymondPhilips** [reply]
/Subscription might find some use, though i'm struggling whether you want to have a different way of interacting with the CDR just for this single endpoint. All other interactions with CDR are from code.

For HDS purposes, I don't believe in the Terraform use case to create Practitioners. That would conflict with our targeted way of working where user management lies in IAM and we have functionality to create a copy of that information into the CDR. Once that is in place (which i think is pretty soon), no Practitioner shall be created from TF as that may cause conflicts/duplicates.

**Reply by loafoe** [reply]
For cases where users are coming into the system through a manual / UI workflow it does not make sense indeed to couple this with Terraform. GitOps driven onboarding could benefit from this approach though. We'll see how it goes, the practitioner resource is ready for release. Thanks again for your feedback!
